src/crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node.h

```diff
@@ -0,0 +1,93 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+
+#pragma once
+
+#include "crimson/common/log.h"
+#include "crimson/os/seastore/seastore_types.h"
+#include "crimson/os/seastore/transaction_manager.h"
+#include "crimson/os/seastore/extentmap_manager.h"
+
+namespace crimson::os::seastore::extentmap_manager{
+
+struct ExtMapNode : LogicalCachedExtent {
+  TransactionManager* tm = nullptr;
+  using ExtMapNodeRef = TCachedExtentRef<ExtMapNode>;
+  depth_t depth = 0;
+
+  ExtMapNode(ceph::bufferptr &&ptr) : LogicalCachedExtent(std::move(ptr)) {}
+  ExtMapNode(const ExtMapNode &other)
+  : LogicalCachedExtent(other),
+    tm(other.tm),
+    depth(other.depth) {}
+
+  void set_depth(depth_t _depth) { depth = _depth; }
+  void set_tm(TransactionManager* _tm) { tm = _tm; }
+
+  LogicalCachedExtentRef make_duplicate(Transaction& t,
+	  LogicalCachedExtentRef ref) {
+    return tm->get_mutable_extent(t, ref);
+  };
+
+  using seek_lextent_ertr = ExtentMapManager::seek_lextent_ertr;
+  using seek_lextent_ret = ExtentMapManager::seek_lextent_ret;
+  virtual seek_lextent_ret seek_lextent(Transaction &t,
+		                        objaddr_t lo, extent_len_t len) = 0;
+  using insert_ertr = TransactionManager::read_extent_ertr;
+  using insert_ret = insert_ertr::future<ExtentRef>;
+  virtual insert_ret insert(Transaction &t, objaddr_t lo, lext_map_val_t val) = 0;
+
+  using punch_lextent_ertr = ExtentMapManager::punch_lextent_ertr;
+  using punch_lextent_ret = ExtentMapManager::punch_lextent_ret;
+  virtual punch_lextent_ret punch_lextent(Transaction &t, objaddr_t lo, extent_len_t len) = 0;
+
+  using find_hole_ertr = TransactionManager::read_extent_ertr;
+  using find_hole_ret = find_hole_ertr::future<ExtentRef>;
+  virtual find_hole_ret find_hole(Transaction &t, objaddr_t lo, extent_len_t len) = 0;
```

Comment:
What is find_hole used for?

Reply:
for example.  write an object from offset 0(object inner address offset) to 200 (length =200), lba allocate at least one page and map to laddr = 4096,  in extentmap there is an entry 0-> (4096, 0 (offset in this page), 200 (length),  next time write object from offset 200 to 800, I think it'd better to  get the first laddr and find a offset in it to map 200->(4096, 500(offset in this page), 600 (length)), needn't ask lba allocate a new page for it.  Do you think it is necessary?

Reply:
Can you write down exactly what the output is?  Actually, it would really help if each of these public methods had a description.